### PR TITLE
Add Intercom widget to demo site

### PR DESCRIPTION
**Pull request by OtterleyW**
Intercom widget will be anonym until user logs. When the user logs in, the widget should get the user's email and id. After the user logs out, the widget will be restarted to clear the message history in the widget. 
st-flex-jenni-ic herokuapp com_

### CODE REVIEW

File: src/components/Page/Page.js [hunk 1]
```diff
@@ -182,6 +186,24 @@ class PageComponent extends Component {
       });
     }
 
+    const userInfoMaybe =
+      isAuthenticated && currentUser
+        ? {
+            email: currentUser.attributes.email,
+            user_id: currentUser.id.uuid,
```
Comment (kpuputti) [inline src/components/Page/Page.js]: Maybe remove this so the duplicate user problem isn't made worse. Make sure to test with an email that you know exists in Intercom, that might produce errors when multiple users are found.

File: src/components/Page/Page.js [hunk 2]
```diff
@@ -182,6 +186,24 @@ class PageComponent extends Component {
...
+        : null;
+
+    const addIntercomWidget = () => {
+      window.Intercom('boot', {
+        app_id: process.env.REACT_APP_INTERCOM_APP_ID,
+        ...userInfoMaybe,
+      });
+    };
+
+    const intercomWidget = typeof window !== 'undefined' ? addIntercomWidget() : null;
```
Comment (kpuputti) [inline src/components/Page/Page.js]: This is a DOM related side effect, should be in `componentDidMount`. There you don't have to check for `window`.

File: src/components/Topbar/Topbar.js [hunk 3]
```diff
@@ -122,6 +122,11 @@ class TopbarComponent extends Component {
         window.location = path;
       }
 
+      // Shutdown Intercom
+      if (typeof window !== 'undefined') {
+        window.Intercom('shutdown');
+        window.Intercom('boot', { app_id: process.env.REACT_APP_INTERCOM_APP_ID });
+      }
```
Comment (kpuputti) [inline src/components/Topbar/Topbar.js]: I think these should happen before the `window.location = path` above. That refreshes the page. It probably executes this code still, but better to handle cleanup first.

File: src/components/Page/Page.js (hunk 1 above)
Comment (OtterleyW) [reply src/components/Page/Page.js]: I tested this and it doesn't work without the user id (same issue as in the Console).

File: src/components/Page/Page.js [hunk 4]
```diff
@@ -207,6 +230,7 @@ class PageComponent extends Component {
         >
           {children}
         </div>
+        {this.state.intercomWidget}
```
Comment (kpuputti) [inline src/components/Page/Page.js]: I don't think this needs to be rendered? I think the whole state object isn't needed. Or am I missing something?

Essentially you have just the `addIntercomWidget` function that injects the Intercom widget to the UI by calling the `window.Intercom` function. That function could even be outside the React component and only called when we want to render the widget.

File: src/components/Page/Page.js [hunk 5]
```diff
@@ -42,8 +45,28 @@ class PageComponent extends Component {
     // handling both dragover and drop events.
     document.addEventListener('dragover', preventDefault);
     document.addEventListener('drop', preventDefault);
+    const intercomWidget = this.addIntercomWidget(this.props);
```
Comment (kpuputti) [inline src/components/Page/Page.js]: Is it possible that the `currentuser` data isn't yet available when the widget is mounted? In that case, you might have to add a [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate) method that calls the `window.Intercom('update', ...)` with the new data.

**Pull request merged**